server/routes/helpers.js

```diff
@@ -88,24 +88,26 @@ const getPaginationObject = (pageParams) => {
   }
 
   for (let i = startNum; i <= endNum; i++) {
+    const href = `/${courtCode}/case/${caseId}/hearing/${hearingId}/match/defendant/${defendantId}?page=${i}`
     pageItems.push({
       text: i,
-      href: `/${courtCode}/case/${caseId}/hearing/${hearingId}/match/defendant/${defendantId}?page=${i}`,
+      href: showAllMatches ? `${href}&showAllMatches=True` : href,
       selected: currentPage === i
     })
   }
 
   if (currentPage !== 1) {
     previousLink = {
       text: 'Previous',
-      href: `/${courtCode}/case/${caseId}/hearing/${hearingId}/match/defendant/${defendantId}?page=${currentPage - 1}`
+      href: `/${courtCode}/case/${caseId}/hearing/${hearingId}/match/defendant/${defendantId}?page=${currentPage - 1}${showAllMatches ? '&showAllMatches=True' : ''}`
     }
   }
 
+  const allMatchesShown = showAllMatches ? '&showAllMatches=True' : ''
```

Comment:
Would be great to move this to line 89 and replace both other uses of `${showAllMatches ? '&showAllMatches=True' : ''}` with `$href$allMatchesShown` or similar

Reply:
The idea is to do the conditional check once only - we are still doing it 3 times. Once we know whether showAllMatches is true, we should not need to ask again for the purposes of building a URL